src/createrepo_c.c

```diff
@@ -1416,7 +1462,7 @@ main(int argc, char **argv)
      * that unfortunately means we have to decompress metadata files change package
      * count value and compress them again.
      */
-    if (user_data.package_count != user_data.task_count){
+    if (user_data.package_count != user_data.task_count - skipped_pkgs){
```

Comment:
The `user_data.task_count` is also used in the [`cr_rewrite_pkg_count_thread()`](https://github.com/rpm-software-management/createrepo_c/blob/master/src/threads.c#L134) which causes problems because the `skipped_pkgs` are not taken into account. 

It can be reproduced by running with invalid packages (such as `echo "test" > bad.rpm`).